clash-vexriscv/src/ffi/impl.cpp

```diff
@@ -127,3 +165,172 @@ void vexr_step_falling_edge(VVexRiscv *top, uint64_t time_add, const COMB_INPUT
 	// Evaluate the simulation
 	top->eval();
 }
+
+vexr_jtag_bridge_data *vexr_jtag_bridge_init(uint16_t port)
+{
+	vexr_jtag_bridge_data *d = new vexr_jtag_bridge_data;
+
+	d->prev_input = { 0, 0, 0 };
+
+	d->tck_change_counter = 0;
+
+	d->timer = 0;
+	d->self_sleep = 0;
+	d->check_new_connections_timer = 0;
+	d->rx_buffer_size = 0;
+	d->rx_buffer_remaining = 0;
+
+	d->server_socket = socket(PF_INET, SOCK_STREAM, 0);
+	assert(d->server_socket != -1);
+	int flag = 1;
+	setsockopt(
+		d->server_socket, /* socket affected */
+		IPPROTO_TCP,                /* set option at TCP level */
+		TCP_NODELAY,                /* name of option */
+		(char *) &flag,             /* the cast is historical cruft */
+		sizeof(int)                 /* length of option value */
+	);
+
+	set_socket_blocking_enabled(d->server_socket, 0);
+
+	d->server_addr.sin_family = AF_INET;
+	d->server_addr.sin_port = htons(port);
+	d->server_addr.sin_addr.s_addr = inet_addr("127.0.0.1");
+	memset(d->server_addr.sin_zero, '\0', sizeof(d->server_addr.sin_zero));
+
+	bind(
+		d->server_socket,
+		(struct sockaddr *) &d->server_addr,
+		sizeof(d->server_addr)
+	);
+
+	listen(d->server_socket, 1);
+
+	d->client_handle = -1;
+	d->addr_size = sizeof(d->server_storage);
+
+	return d;
+}
+
+void vexr_jtag_bridge_step(vexr_jtag_bridge_data *d, const JTAG_OUTPUT *output, JTAG_INPUT *input)
+{
+	const int WAIT_PERIOD = 83333;
```

Comment:
I took this from the blog post version, but this is also not used anywhere, this should be removed.